packages/back-end/src/services/experimentNotifications.ts

```diff
@@ -0,0 +1,232 @@
+    changes: {
+      pastNotifications: [...(experiment.pastNotifications || []), type],
+    },
+  });
+};
+
+export const notifyFailedAutoUpdate = async ({
+  context,
+  experimentId,
+}: {
+  context: Context;
+  experimentId: string;
+}) => {
+  const experiment = await getExperimentById(context, experimentId);
+
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  await updateWrapper({
+    context,
+    experiment,
+    type: "auto-update",
+    handler: () =>
+      dispatchEvent(context, {
+        type: "auto-update",
+        success: false,
+        experimentId,
+        experimentName: experiment.name,
+      }),
+  });
+};
+
+export const notifyAutoUpdateSuccess = async ({
+  context,
+  experimentId,
+}: {
+  context: Context;
+  experimentId: string;
+}) => {
+  const experiment = await getExperimentById(context, experimentId);
+
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  if (!experiment.pastNotifications?.includes("auto-update")) return;
+
+  await dispatchEvent(context, {
+    type: "auto-update",
+    success: true,
+    experimentId,
+    experimentName: experiment.name,
+  });
+
+  await updateExperiment({
+    experiment,
+    context,
+    changes: {
+      pastNotifications: (experiment.pastNotifications || []).filter(
+        (v) => v !== "auto-update"
+      ),
+    },
+  });
+};
+
+export const MINIMUM_MULTIPLE_EXPOSURES_PERCENT = 0.01;
+
+const notifyMultipleExposures = async ({
+  context,
+  experiment,
+  lastResult,
+  snapshot,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+  snapshot: ExperimentSnapshotDocument;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "multiple-exposures",
+    handler: async () => {
+      const totalsUsers = lastResult.variations.reduce(
+        (totalUsersCount, { users }) => totalUsersCount + users,
+        0
+      );
+      const percent = snapshot.multipleExposures / totalsUsers;
+      const multipleExposureMinPercent =
+        context.org.settings?.multipleExposureMinPercent ??
+        MINIMUM_MULTIPLE_EXPOSURES_PERCENT;
+
+      if (snapshot.multipleExposures < multipleExposureMinPercent) return;
+
+      await dispatchEvent(context, {
+        type: "multiple-exposures",
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+        usersCount: snapshot.multipleExposures,
+        percent,
+      });
+    },
+  });
+
+export const DEFAULT_SRM_THRESHOLD = 0.001;
+
+const notifySrm = async ({
+  context,
+  experiment,
+  lastResult,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "srm",
+    handler: async () => {
+      const srmThreshold =
+        context.org.settings?.srmThreshold ?? DEFAULT_SRM_THRESHOLD;
+
+      if (srmThreshold <= lastResult.srm) return;
```

Comment:
Similarly I wonder if this should be in a shared package and used in both the FE and BE as well, akin to the `multiExposureCheck` method. This one is just a < comparison so maybe its an even dumber suggestion here, but I love the idea of the business logic being in one place.